### ISSUE TITLE
no-unused-expressions shouldn't be triggered by Flow "type assertion" lines

### ISSUE THREAD

**Issue by julienw** [eslint/eslint#9877]
Type assertions is something that's [recommended in Flow's documentation](https://flow.org/en/docs/types/casting/#toc-type-assertions).

But sadly this triggers the rule `no-unused-expressions`. I think this shouldn't happen, or it should be made configurable, because the expression has indeed a use.

Example code:
```javascript
("foo": string);
```
.eslintrc.js:
```javascript
module.exports = {
  parser: 'babel-eslint',
  rules: {
    'no-unused-expressions': 'error',
  },
};
```
Eslint version: 4.16.0.

**Comment by platinumazure** [comment]
I don't think this is something ESLint core should support directly. I think a new version of the `no-unused-expressions` rule (which is aware of these semantics) should go into a plugin of some type.

eslint-plugin-babel is normally used for rules that need to be changed/altered to account for semantics of extensions or experimental features, but I'm not sure if they handle Flow-specific rules or if there's a Flow plugin out there where this should go.

**Comment by j-f1** [comment]
There’s [`flowtype/no-unused-expressions`](https://github.com/gajus/eslint-plugin-flowtype#no-unused-expressions), which doesn’t report an error with typecasts like this.

**Comment by julienw** [comment]
I see, thanks for the pointer, my search fu hadn't got me anywhere.
Let's close this bug, hopefully this will show up in searches now.
